services/cd-service/pkg/repository/transformer.go

```diff
@@ -2792,6 +2792,17 @@ func (c *CreateEnvironment) Transform(
 		if err != nil {
 			return "", fmt.Errorf("Unable to write overview cache, error: %w", err)
 		}
+
+		//Should be empty on new environments
+		envApps, err := state.GetEnvironmentApplications(ctx, transaction, c.Environment)
+		if err != nil {
+			return "", fmt.Errorf("Unable to read environment, error: %w", err)
+
+		}
+		for _, app := range envApps {
+			t.AddAppEnv(app, c.Environment, "")
+		}
+
```

Comment:
if envApps should be empty, then why are we doing this look?
I think we can skip it.